test/groovy/TestsPublishResultsTest.groovy

```diff
@@ -126,4 +129,45 @@ class TestsPublishResultsTest extends BasePiperTest {
         assertTrue('Cobertura options are not empty', publisherStepOptions.cobertura == null)
         assertTrue('JMeter options are not empty', publisherStepOptions.jmeter == null)
     }
+
+    @Test
+    void testBuildResultStatus() throws Exception {
+        // execute test
+        stepRule.step.testsPublishResults(
+            script: nullScript
+        )
+        // asserts
+        assertJobStatusSuccess()
+    }
+
+    @Test
+    void testBuildResultStatusWithTestFailures() throws Exception {
```

Comment:
I do not really get this test. In the code there is nothing about test failiures. It basically tests, that if it was unstable before, it is unstable afterwards. 
(I guess you set the build to unstable, because of test failures, but that is not clear from the code or the name of the test.)

Reply:
This test is obsolete as the `currentBuild` is not considered anymore inside the step. I'll remove it.